zipline/utils/cache.py

```diff
@@ -143,8 +174,14 @@ def get(self, key, dt):
         try:
             return self._cache[key].unwrap(dt)
         except Expired:
-            del self._cache[key]
-            raise KeyError(key)
+            # We can't handle safely handle the exception in this block
```

Comment:
If we're going to re-raise an exception, then we don't need to explicitly clear the exception in the cleanup function.  The important thing is just that by the time we exit this function, the exception referencing the cached object is no longer in `sys.exc_info`.